flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -168,6 +171,9 @@
 
 	private final CompletableFuture<Void> closeFuture;
 
+	@Nullable
```

Comment:
optional: add NoopListener such that we do not need to perform null checks.

Reply:
Yeah, i agree with this optional for formal codes. Let's keep it in PoC, and we also had many similar cases before like `RecordWriter#outputFlusher`. :)